src/uptane/tufrepository.cc

```diff
@@ -124,6 +128,13 @@ std::string TufRepository::downloadTarget(Target target) {
 void TufRepository::saveTarget(const Target& target) {
   if (target.length() > 0) {
     downloadTarget(target);
+  } else if (target.format().empty() || target.format() == "OSTREE") {
+#ifdef BUILD_OSTREE
+    data::PackageManagerCredentials cred(CryptoKey(storage_, config_));
+    OstreeManager::pull(config_, cred, target.sha256Hash());
+#else
+    LOG_ERROR << "Could not pull ostree target. Aktualizr builded without OSTree support!";
+#endif
```

Comment:
I think this is punching through a few too many walls. I'd prefer if `pull` were part of the package manager interface so that we don't need another `BUILD_OSTREE` guard here. It would mean `pull` couldn't be static, which is probably fine, and that would mean we wouldn't have to pass down the entire `config` object and could instead just pass the server URL and use the member `pacman` config object. It might mean this call needs to be up some levels. I don't think we want to call directly into OSTree from inside the tuf repo.

Reply:
I agree, I plan to change it in my PR for deb backend